Ltac2: Make it possible to define externals with other plugin names (#17533)
coq/coq: doc/sphinx/proof-engine/ltac2.rst

   Records are product types with named fields and eliminated by projection.
   Likewise they can be recursive if the `rec` flag is set.

.. cmd:: Ltac2 @ external @ident : @ltac2_type := @string__plugin @string__function
   :name: Ltac2 external

   Declares functions defined in OCaml. :n:`@string__plugin` is the

>    plugin name defining the function.  :n:`@string_function` is the internal name

Reviewer: ```suggestion
   plugin name defining the function.  :n:`@string__function` is the internal name
```

Must be two underscores.

> .. cmd:: Ltac2 @ external @ident : @ltac2_type := @string @string
>    :name: Ltac2 external

Reviewer: ```suggestion
.. cmd:: Ltac2 @ external @ident : @ltac2_type := @string__plugin @string__function
   :name: Ltac2 external
```

>    Declares functions defined in OCaml. The first :n:`@string` is the
>    plugin name defining the function, the second is the internal name

Reviewer: ```suggestion
   Declares functions defined in OCaml. :n:`@string__plugin` is the
   plugin name defining the function.  :n:`@string_function` is the internal name
```

Subscripts are your friend.

>    of the function. As a special case, ``"ltac2"`` stands for plugin name

Reviewer: ```suggestion
   of the function. As a special case, ``"ltac2"`` represents the plugin name
```
